boards/esp32-ethernet-kit/doc.txt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ * Copyright (C) 2020 Google LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    boards_esp32_esp-ethernet-kit ESP32-Ethernet-Kit
+ * @ingroup     boards_esp32
+ * @brief       Support for for Espressif ESP32-Ethernet-Kit V1.1+
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @author      Erik Ekman <eekman@google.com>
+
+## <a name="toc"> Table of Contents </a>
+
+1. [Overview](#overview)
+2. [Hardware](#hardware)
+    1. [MCU](#mcu)
+    2. [Board Configuration](#board_configuration)
+    3. [Board Pinout](#pinout)
+3. [Flashing the Device](#flashing)
+4. [On-Chip Debugging with the device](#debugging)
+5. [Other Documentation Resources](#other-resources)
+
+## <a name="overview"> Overview </a> &nbsp;&nbsp; [[TOC](#toc)]
+
+The Espressif ESP32-Ethernet-Kit is a development board that uses the ESP32-WROVER-B (v1.0, v1.1)/ESP32-WROVER-E (v1.2) module. Most important features of the board are
+
+- 100 Mbps Ethernet via IP101G Phy
+- USB bridge with JTAG interface
+
+Furthermore, some GPIOs are broken out for extension. The USB bridge based on FDI FT2232HL provides a JTAG interface for OCD debugging through the USB interface.
+
+## <a name="hardware"> Hardware </a> &nbsp;&nbsp; [[TOC](#toc)]
+
+This section describes
+
+- the [MCU](#mcu),
+- the default [board configuration](#board_configuration),
+- the [board pinout](#pinout).
+
+### <a name="mcu"> MCU </a> &nbsp;&nbsp; [[TOC](#toc)]
+
+Most features of the board are provided by the ESP32 SoC. For detailed
+information about the ESP32, see section \ref esp32_mcu "MCU ESP32".
+
+### <a name="board_configuration"> Board Configuration </a> &nbsp;&nbsp; [[TOC](#toc)]
+
+ESP32-Ethernet-Kit has the following on-board components
+
+- 100 Mbps Ethernet via IP101G Phy
+- USB bridge with JTAG interface
+
+For detailed information about the configuration of ESP32 boards, see
+section Peripherals in \ref esp32_riot.
+
+https://docs.espressif.com/projects/esp-idf/en/latest/esp32/hw-reference/esp32/get-started-ethernet-kit.html
+
+@note
+- Only a few GPIOs are available for external hardware on ESP32-Ethernet-Kit boards.
```

Comment:
It would be great to have a table with the default board configuration as for other boards here. See for example `esp32-wroom-32`.

```suggestion
Only a few GPIOs are broken out and available for external hardware on ESP32-Ethernet-Kit boards. Which GPIOs are available as peripherals depends on used modules.

<center>
\anchor esp32_ethernet_kit_table_board_configuration
Function        | GPIOs  | Remarks |Configuration
:---------------|:-------|:--------|:----------------------------------
BTN0            | GPIO0  | not available if `esp_eth` is used | |
ADC             | GPIO34, GPIO35, GPIO36, GPIO39 | | see \ref esp32_adc_channels "ADC Channels"
DAC             | - | | \ref esp32_dac_channels "refer"
PWM_DEV(0)      | GPIO2, GPIO4 | GPIO2 is not available in V1.0 | \ref esp32_pwm_channels "PWM Channels"
I2C_DEV(0):SDA  | GPIO32 | | \ref esp32_i2c_interfaces "I2C Interfaces"
I2C_DEV(0):SCL  | GPIO33 | | \ref esp32_i2c_interfaces "I2C Interfaces"
SPI_DEV(0):CLK  | GPIO14 | HSPI is used | \ref esp32_spi_interfaces "SPI Interfaces"
SPI_DEV(0):MISO | GPIO12 | HSPI is used | \ref esp32_spi_interfaces "SPI Interfaces"
SPI_DEV(0):MOSI | GPIO13 | HSPI is used | \ref esp32_spi_interfaces "SPI Interfaces"
SPI_DEV(0):CS0  | GPIO15  | HSPI is used | \ref esp32_spi_interfaces "SPI Interfaces"
UART_DEV(0):TxD | GPIO1  | Console (configuration is fixed) | \ref esp32_uart_interfaces "UART interfaces"
UART_DEV(0):RxD | GPIO3  | Console (configuration is fixed) | \ref esp32_uart_interfaces "UART interfaces"
</center>

@note
- GPIO2 is not avialable as PWM channel on ESP32-Ethernet-Kit V1.0.
- GPIO4 works only properl on ESP32-Ethernet-Kit V1.0 board if the function switch for GPIO4 (DIP SW 5) is OFF.
- SPI_DEV(0) is not avialable if module `esp_jtag` is used. For the SPI_DEV(0) pins to work properly, the function switches (DIP switches) for the JTAG signals must be set to OFF.
```